### PR TITLE
Use `FlameTester` everywhere where it makes sense

### LINKED ISSUE #946: Define a standard for casing on test names
# What could be improved
We should define a standard for casing on test and test group names in our test files.

# Why should this be improved
Since it is a mix now, and someone might start crying when reading the tests.

# Any risks?
No.

# More information

### PR DESCRIPTION
**Pull request by spydon**
# Description
Uses the `FlameTester` everywhere where it makes sense and unifies the style a little bit on the tests (there is still a lot to do and we need to set up some standards).

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

Closes #946 

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame_test/lib/src/flame_test.dart [hunk 1]
```diff
@@ -12,7 +12,7 @@ extension FlameFinds on CommonFinders {
   }
 }
 
-extension FlameGameExtension on FlameGame {
+extension FlameGameExtension on Component {
```
Comment (luanpotter) [inline packages/flame_test/lib/src/flame_test.dart:15]: should this be marked visibleForTest? or being on flame-test is enough?

File: packages/flame/test/game/projections_test.dart [hunk 2]
```diff
@@ -194,9 +207,10 @@ void main() {
       expect(game.projector.unscaleVector(Vector2.zero()), Vector2.zero());
       expect(game.projector.unscaleVector(Vector2(2, 4)), Vector2(1, 2));
     });
-    test('camera & viewport - two translations', () {
+
+    flameGame.test('camera & viewport - two translations', (game) {
       final viewport = FixedResolutionViewport(Vector2.all(100));
-      final game = FlameGame()..camera.viewport = viewport; // default camera
+      game..camera.viewport = viewport; // default camera
```
Comment (luanpotter) [inline packages/flame/test/game/projections_test.dart:213]:
```suggestion
      game.camera.viewport = viewport; // default camera
```
(and elsewhere)

File: packages/flame/test/components/tappables_test.dart [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
-        throwsA(
-          predicate(
-            (e) => e is AssertionError && e.message == message,
+        expect(
+          () => game.add(_TappableComponent()),
+          throwsA(
+            predicate(
+              (e) => e is AssertionError && e.message == message,
+            ),
           ),
```
Comment (luanpotter) [inline packages/flame/test/components/tappables_test.dart:34]: we should probablty have a test helper to make it simple to test exceptions with messages. testing just any exception is thrown is quite fragile

File: packages/flame/test/components/sprite_animation_component_test.dart [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
   final image = await generateImage();
 
-  final size = Vector2(1.0, 1.0);
-
-  group('SpriteAnimationComponent shouldRemove test', () {
-    test('removeOnFinish is true and animation#loop is false', () {
-      final game = FlameGame();
+  group('SpriteAnimationComponent shouldRemove', () {
+    flameGame.test('removeOnFinish is true and animation#loop is false',
+        (game) async {
```
Comment (luanpotter) [inline packages/flame/test/components/sprite_animation_component_test.dart:11]: formatting!

Comment (spydon) [reply packages/flame/test/components/sprite_animation_component_test.dart:11]: Why isn't the linter complaining?!

File: packages/flame/test/components/tappables_test.dart (hunk 3 above)
Comment (spydon) [reply packages/flame/test/components/tappables_test.dart:34]: We should indeed, unrelated to this PR though.

File: packages/flame/test/components/has_game_ref_test.dart [hunk 5]
```diff
@@ -2,24 +2,24 @@ import 'package:flame/components.dart';
...
+class _MyComponent extends Component with HasGameRef<_MyGame> {
   void foo() {
     gameRef.foo();
   }
 }
 
 void main() {
-  group('has game ref test', () {
+  group('HasGameRef', () {
     test('simple test', () {
```
Comment (luanpotter) [inline packages/flame/test/components/has_game_ref_test.dart:20]: why not flame tester?

File: packages/flame/test/components/draggable_test.dart [hunk 6]
```diff
@@ -19,42 +18,42 @@ class DraggableComponent extends PositionComponent with Draggable {
...
+        const message =
+            'Draggable Components can only be added to a FlameGame with '
+            'HasDraggableComponents';
+
+        expect(
+          () => game.add(_DraggableComponent()),
+          throwsA(
+            predicate(
+              (e) => e is AssertionError && e.message == message,
+            ),
```
Comment (luanpotter) [inline packages/flame/test/components/draggable_test.dart:43]: other example

**Review by luanpotter — APPROVED** [review]
wow! LGTM. Thanks for putting this together @spydon !!!

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**